### DISCUSSION TITLE
Is there a way to get the name of the dir in which a file is being created?

### DISCUSSION THREAD

**Discussion by AndOrangutan** [zk-org/zk#191] [Help] [answered]
To explain the title, I want to be able to use the name of the directory in which a file was created as one of its tags. 

Lets say I wanted to create a note in NOTEBOOK/classes/computersciece/new-note.md

```bash
compendium > tree          
.
├── classes
│   ├── classes.md
│   └── computerscience
│       └── computer-science.md
└── compendium.md
```
I would want to set up note template that looked something like this.

```markdown
---
date: {{date now "%F %T"}}
keywords: [note, {{DIRNAME}}]
---

# {{title}}

{{content}}
```

Which in my example would look something like:
```markdown
---
date: 2022-04-20 21:47:39
keywords: [index, computerscience]
---

# New Note
```

If I could somehow do the parent of that directory too would be useful too:

```markdown
---
date: 2022-04-20 21:47:39
keywords: [index, classes, computerscience]
---

# New Note
```

Thanks :D

**Comment by mickael-menu** [comment, marked as answer]
Hi,

It's not possible, but you have access to the `{{dir}}` variable which would return `classes/computerscience` in your case.

Not perfect, [but you can use wildcards when searching for tags](https://github.com/mickael-menu/zk/blob/main/docs/note-filtering.md#filter-by-tags).

I don't want to add more variables for use cases that are too specific, but I think the `sh` template helpers could be extended to support passing template variables. This would open a lot of possibilities and you could use something like this in your case:

```
keywords: [note, {{sh 'basename {{dir}}'}}]
```
